plugins/filesystem/local/adapter/adapter.php

```diff
@@ -326,4 +326,278 @@ private function getDate($date = null)
+			throw new Exception('Copy is not possible');
+		}
+	}
+
+	/**
+	 * Moves a file or folder to a destination
+	 * If the destination folder or file already exists, it will not overwrite them without
+	 * force.
+	 *
+	 * @param   string  $sourcePath       Source path of the file or directory
+	 * @param   string  $destinationPath  Destination path of the file or directory
+	 * @param   bool    $force            Set true to overwrite files or directories
+	 *
+	 * @return void
+	 *
+	 * @since __DEPLOY_VERSION__
+	 * @throws MediaFileAdapterFilenotfoundexception
+	 */
+	public function move($sourcePath, $destinationPath, $force = false)
+	{
+		// Get absolute paths from relative paths
+		$sourcePath = $this->rootPath . $sourcePath;
+		$destinationPath = $this->rootPath . $destinationPath;
+
+		if (!file_exists($sourcePath))
+		{
+			throw new MediaFileAdapterFilenotfoundexception;
+		}
+
+		if (is_dir($sourcePath))
+		{
+			$this->moveFolder($sourcePath, $destinationPath, $force);
+		}
+		else
+		{
+			$this->moveFile($sourcePath, $destinationPath, $force);
+		}
+	}
+
+	/**
+	 * Moves a file
+	 *
+	 * @param   string  $sourcePath       Absolute path of source
+	 * @param   string  $destinationPath  Absolute path of destination
+	 * @param   bool    $force            Set true to overwrite file if exists
+	 *
+	 * @return void
+	 *
+	 * @since __DEPLOY_VERSION__
+	 * @throws  Exception
+	 */
+	protected function moveFile($sourcePath, $destinationPath, $force = false)
+	{
+		$success = false;
+		if (!file_exists($destinationPath))
+		{
+			$success  = JFile::move($sourcePath, $destinationPath);
+		}
+		else
+		{
+			if ($force && !is_dir($destinationPath))
+			{
+				$success  = JFile::move($sourcePath, $destinationPath);
+			}
+		}
+
+		if (!$success)
+		{
+			throw new Exception('Move is not possible');
+		}
+	}
+
+	/**
+	 * Moves a folder from source to destination
+	 *
+	 * @param   string  $sourcePath       Source path of the file or directory
+	 * @param   string  $destinationPath  Destination path of the file or directory
+	 * @param   bool    $force            Set true to overwrite files or directories
+	 *
+	 * @return void
+	 *
+	 * @since __DEPLOY_VERSION__
+	 * @throws  Exception
+	 */
+	protected function moveFolder($sourcePath, $destinationPath, $force = false)
+	{
+		if (file_exists($destinationPath))
+		{
+			if (is_dir($destinationPath))
+			{
+				// We need to bypass exception thrown in JFolder when destination exists
+				// So we only copy it in forced condition, then delete the source to simulate a move
+
+				$success = false;
+				if ($force)
+				{
+					$copySuccess = JFolder::copy($sourcePath, $destinationPath, '', true);
+					$deleteSuccess = false;
+
+					if ($copySuccess)
+					{
+						$deleteSuccess = JFolder::delete($sourcePath);
+					}
+					else
+					{
+						// Undo previous copy
+						JFolder::delete($destinationPath);
+					}
+
+					$success = $copySuccess && $deleteSuccess;
+				}
+
+				if (!$success)
```

Comment:
in value === false we check for explicit false for the statement, so when a text was returned we can distinguish, i can change !success to sucsess == false to unify if wanted